java-dynamic-sqs-listener-core/src/main/java/com/jashmore/sqs/util/Immutables.java

```diff
@@ -0,0 +1,66 @@
+package com.jashmore.sqs.util;
+
+import static java.util.stream.Collectors.collectingAndThen;
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.Collectors.toSet;
+
+import lombok.experimental.UtilityClass;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Utility class used to provide helper methods to construct Immutable implementations of data structures.
+ *
+ * <p>This has been implemented again to reduce dependencies on other modules, more specifically Guava.
```

Comment:
I would include Guava in favor of doing this - same for Preconditions.

Reply:
Yeah, do you forsee any problems with me having guava 19 and someone is using guava 27 or whatever in their project? I just want to make sure that usage of this isn't a problem.